packages/cmf/src/middlewares/http/middleware.js

```diff
@@ -1,12 +1,81 @@
 import has from 'lodash/has';
 import get from 'lodash/get';
+import merge from 'lodash/merge';
+import flow from 'lodash/flow';
+import curry from 'lodash/curry';
 import { HTTP_METHODS, HTTP_REQUEST, HTTP_RESPONSE, HTTP_ERRORS } from './constants';
 
+/**
+ * @typedef {Object} Action
+ * @property {string} type - Action type
+ */
+
+/**
+ * @typedef {Object} Response
+ * @property {string} status
+ * @property {string} statusText
+ * @property {string} ok
+ * @property {string} redirected
+ * @property {string} type
+ * @property {string} url
+ */
+
+/**
+ * @typedef {Object} Stack
+ * @property {Object} response
+ * @property {Object} message
+ */
+
+/**
+ * @typedef {Object} HttpError
+ * @property {string} name
+ * @property {string} message
+ * @property {string} number
+ * @property {Stack} stack
+*/
+
+/**
+ * @typedef {Object.<string, string>} Headers
+ */
+
+/**
+ * @callback onError
+ * @param {HTTPConfig} action
+ * @param {HttpError} error
+ * @return {Action}
+ */
+
+/**
+ * @callback onResponse
+ * @param {HTTPConfig} action
+ * @param {Object} response
+ * @return {Action}
+ */
+
+/**
+ * @typedef {Object} HTTPConfig
+ * @property {string} body
+ * @property {string} credentials
+ * @property {Headers} headers
+ * @property {string} method - See ./constants.js for a list of suitable method
+ * @property {onError | string} onError
+ * @property {onResponse | string} onResponse
+ * @property {string} onSend - a redux action type
+ */
+
```

Comment:
That's waaaay to much JSDoc. It overloads the code. We need to improve its verbosity.